js/src/Themes.js

```diff
@@ -50,10 +55,35 @@ export class ThemeModel extends WidgetModel {
     } else if (document.body.classList.contains("theme-light")) {
       this.set("dark", false);
       this.save_changes();
+    } else if (window.Jupyter) {
+      // Special case for Jupyter Notebook
+      this.set("dark", false);
+      this.save_changes();
+    } else if (document.body.dataset.vscodeThemeKind === "vscode-dark") {
+      // Special case for VS Code
+      vuetify.framework.theme.dark = true;
+      this.set("dark", true);
+      this.save_changes();
+    } else if (document.body.dataset.vscodeThemeKind === "vscode-light") {
+      vuetify.framework.theme.dark = false;
+      this.set("dark", false);
+      this.save_changes();
+    } else if (document.documentElement.matches("[theme=dark]")) {
+      vuetify.framework.theme.dark = true;
+      this.set("dark_effective", true);
+      this.save_changes();
+    } else if (document.documentElement.matches("[theme=light]")) {
+      vuetify.framework.theme.dark = false;
+      this.set("dark_effective", false);
+      this.save_changes();
```

Comment:
what system does this correspond to?

Reply:
This should work in colab (not that I've been able to test it)